cloudinit/cmd/clean.py

```diff
@@ -126,7 +126,8 @@ def remove_artifacts(init, remove_logs, remove_seed=False, remove_config=None):
     init.read_cfg()
     if remove_logs:
         for log_file in get_config_logfiles(init.cfg):
-            del_file(log_file)
+            if os.path.isfile(log_file):
```

Comment:
This will change behavior for anyone that is using a named pipe as a destination for their logs - the pipe will no longer be cleaned up.